File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):  
        n=len(prices)
        minprices=prices[0]
        maxprices=0
        for i in range(n):
            minprices=min(minprices,prices[i])
            profit=prices[i]-minprices
            maxprices=max(maxprices,prices[i]-minprices) 
        return maxprices

        """
        :type prices: List[int]
        :rtype: int
        """
        

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n=prices.size();
        int minprices=prices[0];
        int maxprices=0;
        for(int i=0;i<n;i++){
            minprices=min(minprices,prices[i]);
            
            maxprices=max(maxprices,prices[i]-minprices);
        }
        return maxprices;
    }
};